### PR TITLE
License Resource Interface Implementation

### PR DESCRIPTION
**Pull request by lehmannk**
Based on the new commons resource abstraction this commit contains the
license interface for confluence as well as adapted unit tests.

### CODE REVIEW

File: src/main/java/de/aservo/atlassian/confluence/confapi/service/LicensesServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        //set licenses
+        licensesBean.getLicenses().forEach(this::setLicense);
+        return getLicenses();
+    }
+
+    @Override
+    public LicensesBean setLicense(LicenseBean licenseBean) {
+        try {
+            licenseHandler.addProductLicense(DEFAULT_LICENSE_REGISTRY_KEY, licenseBean.getKey());
+        } catch (InvalidOperationException e) {
```
Comment (lehmannk) [inline src/main/java/de/aservo/atlassian/confluence/confapi/service/LicensesServiceImpl.java:58]: change to ServerException

File: src/test/java/com/atlassian/confluence/settings/setup/DefaultSingleProductLicenseDetailsView.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
+package com.atlassian.confluence.settings.setup;
+
+import com.atlassian.sal.api.license.SingleProductLicenseDetailsView;
+import de.aservo.atlassian.confapi.model.LicenseBean;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Date;
+
+public class DefaultSingleProductLicenseDetailsView implements SingleProductLicenseDetailsView {
```
Comment (lehmannk) [inline src/test/java/com/atlassian/confluence/settings/setup/DefaultSingleProductLicenseDetailsView.java:10]: check if we find an alternative in sal.api

File: src/main/java/de/aservo/atlassian/confluence/confapi/service/LicensesServiceImpl.java (hunk 1 above)
Comment (lehmannk) [reply src/main/java/de/aservo/atlassian/confluence/confapi/service/LicensesServiceImpl.java:59]: done, is now `InternalServerErrorException`

File: src/test/java/com/atlassian/confluence/settings/setup/DefaultSingleProductLicenseDetailsView.java (hunk 2 above)
Comment (pathob) [reply src/test/java/com/atlassian/confluence/settings/setup/DefaultSingleProductLicenseDetailsView.java:10]: Es gibt eine Impl aber nicht in Atlassian-Public, sodass das so bleiben kann.

**Pull request merged**